exonum-java-binding/core/rust/src/cmd/supervisor_mode.rs

```diff
@@ -0,0 +1,53 @@
+use failure::{self, format_err};
+
+use std::str::FromStr;
+
+/// Mode of the Supervisor service.
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+pub enum SupervisorMode {
+    /// Simple mode.
+    Simple,
+    /// Decentralized mode.
+    Decentralized,
+}
+
+impl FromStr for SupervisorMode {
+    type Err = failure::Error;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        match input.to_ascii_lowercase().as_str() {
+            "simple" => Ok(SupervisorMode::Simple),
+            "decentralized" => Ok(SupervisorMode::Decentralized),
+            _ => Err(format_err!(
+                "Invalid supervisor mode. Can be \"simple\" or \"decentralized\""
```

Comment:
`Invalid supervisor mode: <actual>. ...`

Can also use single quotation marks ' to not have to escape them.